doc/replication-policies.md

```diff
@@ -0,0 +1,195 @@
+for the completion of all of them.
+
+For example:
+```objc
+/** Start both a push and pull replication.
+ *
+ *  @param taskGroup A dispatch_group_t to allow us to wait for replications to complete.
+ */
+- (void)startReplications:(dispatch_group_t)taskGroup
+{
+    // Create and start the replicator -- -start is essential!
+    CDTReplicatorFactory *replicatorFactory =
+    [[CDTReplicatorFactory alloc] initWithDatastoreManager:manager];
+
+    // username/password can be Cloudant API keys
+    NSString *s = @"https://username:password@username.cloudant.com/my_database";
+    NSURL *remoteDatabaseURL = [NSURL URLWithString:s];
+    CDTDatastore *datastore = [manager datastoreNamed:@"my_datastore"];
+
+    // Create a replicator that replicates changes from the local
+    // datastore to the remote database.
+    CDTPushReplication *pushReplication = [CDTPushReplication replicationWithSource:datastore
+                                                                             target:remoteDatabaseURL];
+
+    // Create a replicator that replicates changes from a remote
+    // database to the local one.
+    CDTPullReplication *pullReplication = [CDTPullReplication replicationWithSource:remoteDatabaseURL
+                                                                             target:datastore];
+
+    NSError *pushError;
+    CDTReplicator *pushReplicator = [replicatorFactory oneWay:pushReplication error:&error];
+    pushReplicator.sessionConfigDelegate = self;
+
+    NSError *pullError;
+    CDTReplicator *pullReplicator = [replicatorFactory oneWay:pullReplication error:&error];
+    pullReplicator.sessionConfigDelegate = self;
+
+    // Start the push replication
+    if (![pushReplicator startWithError:&error taskGroup:taskGroup]){
+        //handle error
+    }
+
+    // Start the pull replication
+    if (![pullReplicator startWithError:&error taskGroup:taskGroup]){
+        //handle error
+    }
+}
+```
+
+Create a method to start replications and wait for their completion (on a background thread).
+
+```objc
+-(void)syncInBackgroundWithCompletionHandler:(void (^)())completionHandler
+{
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
+        dispatch_group_t backgroundTasks = dispatch_group_create();
+
+        UIBackgroundTaskIdentifier taskId = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:NULL];
+
+        [self startReplications:backgroundTasks];
+
+        dispatch_group_wait(backgroundTasks, dispatch_time(DISPATCH_TIME_NOW, 30000000000));
+
+        if (completionHandler) {
+            completionHandler();
+        }
+
+        [[UIApplication sharedApplication] endBackgroundTask:taskId];
+    });
+}
+```
+
+Setup an NSURLSessionConfigurationDelegate to customise the NSURLSession as you require:
+
+```objc
+- (NSURLSessionConfiguration*)customiseNSURLSessionConfiguration:(nonnull NSURLSessionConfiguration *)config {
```

Comment:
The delegate is passed to the `CDTReplicator` by setting the delegate, in the example above,there are calls to `pushReplicator.sessionConfigDelegate = self;` and `pullReplicator.sessionConfigDelegate = self;`. Maybe it would be clearer to add the delegate to the factory method creating the `CDTReplicator` making it `oneway:sessionConfigDelegate:error:` instead of `oneway:error:`. Any thoughts?